src/lib/common/include/sol-platform.h

```diff
@@ -236,6 +256,155 @@ int sol_platform_add_hostname_monitor(void (*cb)(void *data, const char *hostnam
 int sol_platform_del_hostname_monitor(void (*cb)(void *data, const char *hostname), const void *data);
 
 /**
+ * @brief Set the system wide time.
+ *
+ * @param timestamp The new system_clock
+ * @return @c 0 on succes, negative errno otherwise
+ *
+ * @note The @c timestamp is relative to 1970-01-01 00:00:00 +0000 (UTC)
+ * @see sol_platform_get_system_clock()
+ */
+int sol_platform_set_system_clock(int64_t timestamp);
```

Comment:
it is fine, this is just used to save calling `time()` when time changes, if more resolution is wanted the user can call something else, like `clock_gettime()`